R/azimuth.R

```diff
@@ -206,10 +206,19 @@ RunAzimuth.Seurat <- function(
       reduction.key = 'UMAP_',
       verbose = verbose
     )
-    # Calculate mapping score and add to metadata
+    
+    # for multi-layered queries, MappingScore() returns a list of layers and associated scores
+    metadata = MappingScore(anchors = anchors, ndim = dims) 
+    metadata_combined <- do.call(rbind, lapply(seq_along(metadata), function(i) {
+      data.frame(
+        score = metadata[[i]], 
+        layer = i  
+      )
+    }))
```

Comment:
Firstly, you can just use `unlist` to flatten lists out. Secondly, you want to make that call inside `Seurat::MappingScore` (more explanation to come in my review of the relevant PR).